staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -50,7 +50,7 @@ import (
 
 var (
 	printerColumnDatatypes                = sets.NewString("integer", "number", "string", "boolean", "date")
-	customResourceColumnDefinitionFormats = sets.NewString("int32", "int64", "float", "double", "byte", "date", "date-time", "password")
+	customResourceColumnDefinitionFormats = sets.NewString("int32", "int64", "float", "double", "byte", "date", "date-time", "password", "k8s-short-name", "k8s-long-name")
```

Comment:
What is the format on column definitions used for?

The API doc isn't very clear and also references a `name` format that doesn't appear to be allowed here? Are these  columns appended to the standard `name` column? If so, it's pretty weird for these new name formats to not be usable on the name column, and for the `name` printer column format to have no openapi format meaning :-/

```diff
	// format is an optional OpenAPI type definition for this column. The 'name' format is applied
	// to the primary identifier column to assist in clients identifying column is the resource name.
	// See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types for details.
	// +optional
	Format string `json:"format,omitempty" protobuf:"bytes,3,opt,name=format"`
```

Reply:
> What is the format on column definitions used for?

I think I misunderstood this.  It looks like this is the `customResourceColumnDefinitionFormats` enumerates the allows formats that a printer column may be declared as.  

I has mistaken this to be a filter on what fields are allowed to used as printer columns.  I've dropped the k8s names from this list and added tests showing that fields using these formats can be used as string printer columns.

I believe this is consistent with the existing behavior and doesn't add any new wrinkles.  I do fine the `name` format odd here since this format field otherwise makes consistent use of OpenAPI names. I'll think about that more.